### ISSUE TITLE
Unable to grant gmvault access

### ISSUE THREAD

**Issue by nowox** [gaubert/gmvault#282]
Right after this message I opened my browser, copied the link then pasted it on my terminal. 

```
https://accounts.google.com/o/oauth2/auth?client_id=154[...]77-0esd[...]fo8esfi0mkrtog.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code&scope=https%3A%2F%2Fmail.google.com%2F

Once you've granted gmvault access, enter the verification code and press enter:
```

Unfortunately the `enter` key doesn't do anything and I am stuck in this mode. After pressing Ctrl+C I am back to my shell and the echo is still off. 

I have Gmvault v1.9.1

**Comment by gaubert** [comment]
You have to copy the link in your browser URL box and click enter. Then you will get a generate token that you have to copy in the terminal and press enter

**Comment by gaubert** [comment]
user mistake

**Comment by nowox** [comment]
Well, not really a user mistake since I exactly did what you said:

1. `$ gmvault sync foobar@gmail.com`
2. Copy/Paste the link in Chrome
3. Authorized gmvault
4. Copy the authentication code
5. Pasted it in gmvault terminal
6. Doomed.

**Comment by gaubert** [comment]
ok on which plateform are you ? (Windows, Mac OSX, Linux ?).
You also need to send me the console output such that I can understand what is going on and potentially reproduce it. Please also activate the debug with --debug
$>gmvault sync foo.bar@gmail.com --debug

**Comment by nowox** [comment]
Not much information though. 

```
$ gmvault sync foobar@gmail.com --debug
Odd. Use default python2.7.x or 2.6.x distribution.

Use gmvault-db located in /home/foobar/gmvault-db.

Activate Gmvault db cleaning.
Activate debugging information.
Debugging logs are going to be saved in file /home/foobar/gmvault.log.

no ENV variable $GMVAULT_DIR defined. Set by default $GMVAULT_DIR to $HOME/.gmvault (/home/foobar/.gmvault)
/home/foobar/.gmvault/gmvault_defaults.conf with version 1.9.1 is too old, overwrite it with the latest file.
Create defaults in /home/foobar/.gmvault/gmvault_defaults.conf. Please touch this file only if you know what to do.
Load defaults from /home/foobar/.gmvault/gmvault_defaults.conf
Authentication performed with Gmail OAuth2 access token.

Initiate interactive session to get OAuth2 token from Gmail.

gmvault will now open a web browser page in order for you to grant gmvault access to your Gmail.
Please make sure you're logged into the correct Gmail account (foobar@gmail.com) before granting access.
Press ENTER to open the browser.
You should now see the web page on your browser now.
If you don't, you can manually open:

https://accounts.google.com/o/oauth2/auth?client_id=1[...]3777-0eecrado[....]krtog.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code&scope=https%3A%2F%2Fmail.google.com%2F

Once you've granted gmvault access, enter the verification code and press enter:
```

From this point I can type anything (even the correct code) and press enter and nothing happens.

**Comment by gaubert** [comment]
What plateform ? Linux ? Did you change the starting script ?
What python version ?

You should also send me via email the /home/foobar/gmvault.log log file such that I can understand why you cannot enter any character ?

If you are on Linux what terminal and shell are you using ?

Thks

**Comment by gaubert** [comment]
No answer ? I will close it in a couple of days if I don't get any support to solve the issue. Thks

**Comment by nowox** [comment]
I'm late :) sorry about this. We got a huge bug at work and I don't have
free time to geek when I am back home. Hopefully I'll get some free time
this weekend.

**Comment by nowox** [comment]
Hey, 

I have a good news for gmvault. the issue is not gmvault's fault. I can reproduce the same problem with this script: 

```
import webbrowser
webbrowser.open('http://www.google.com')
raw_input('Can you head me? ')
```

As a workaround I can just manually get the permission_url and it works. 

Now I have to understand why webbrowser suppresses the echo. This is very weird. 

Thanks anyway.

**Comment by gaubert** [comment]
For the missing outputs, I found an issue with the buffering of stdin,
stdout, stderr. I solved it in v1.9.2 that I am currently testing.
The outputs are not printed but you can still press enter, the browser is
launched and you can then enter the code.
Anyway this issue should hopefully be solved in 1.9.2 when I release it in
a couple of weeks.

**Comment by gaubert** [comment]
@nowox

You can test the latest beta version here (beta windows and mac os x package available):
https://www.dropbox.com/sh/d5ceo77juacr03y/AACUGcTt6Um-6j6JmBizGPA2a?dl=0

or from the branch gmv-1.9.2

Help required for testing to see if I missed something. Many thanks for the testing.

Let me know.
